doc/cli_subprocess_guidelines.md

```diff
@@ -0,0 +1,110 @@
+# Azure CLI Subprocess Guidelines
+
+In certain cli modules, there are scenarios that need to call a subsystem to run commands outside cli, like getting kubectl info in aks, or deployment setup in mysql using `git` and `gh`, through python built-in [subprocess](https://docs.python.org/3/library/subprocess.html) module. Despite its simplicity and versatility, ensuring the security of applying it can be challenging in different platforms under different circumstance, and it is error-prone for developers to neglect security best practices during development.
```

Comment:
Why using subsystem instead of subprocess? I don't think "subsystem" stands for the scenario this document is talking about.

https://docs.python.org/3/library/subprocess.html doesn't mention the term "subsystem" at all.

Reply:
Will it be clearer to use "subprocess" instead of "subsystem"? I don't think "subsystem" stands for the scenario this document is talking about.

https://docs.python.org/3/library/subprocess.html doesn't mention the term "subsystem" at all.